attestation-container/attest/attestation.go

```diff
@@ -0,0 +1,150 @@
+package attest
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+// Data structures are based on SEV-SNP Firmware ABI Specification
+// https://www.amd.com/en/support/tech-docs/sev-secure-nested-paging-firmware-abi-specification
+
+const (
+	ATTESTATION_REPORT_SIZE = 1184 // Size of ATTESTATION_REPORT (Table 21)
+	REPORT_DATA_SIZE        = 64   // Size of REPORT_DATA_SIZE in ATTESTATION_REPORT
+	REPORT_REQ_SIZE         = 96   // Size of MSG_REPORT_REQ (Table 20)
+	REPORT_RSP_SIZE         = 1280 // Size of MSG_REPORT_RSP (Table 23)
+	PAYLOAD_SIZE            = 40   // Size of sev_snp_guest_request struct from sev-snp driver include/uapi/linux/psp-sev-guest.h
+)
+
+// Message Type Encodings (Table 100)
+const (
+	MSG_REPORT_REQ = 5
+	MSG_REPORT_RSP = 6
+)
+
+// From sev-snp driver include/uapi/linux/psp-sev-guest.h
+const SEV_SNP_GUEST_MSG_REPORT = 3223868161
+
+const SNP_DEVICE_PATH = "/dev/sev"
+
+/*
+Creates and returns MSG_REPORT_REQ message bytes (SEV-SNP Firmware ABI Specification Table 20)
+*/
+func createReportReqBytes(reportData [REPORT_DATA_SIZE]byte) [REPORT_REQ_SIZE]byte {
+	reportReqBytes := [REPORT_REQ_SIZE]byte{}
+	for i := 0; i < REPORT_DATA_SIZE; i++ {
+		reportReqBytes[i] = reportData[i]
+	}
```

Comment:
Is it possible to use `copy()` here instead?

Reply:
Addressed in 263280d